app/uk/gov/hmrc/individualincomedesstub/domain/SelfEmployment.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.individualincomedesstub.domain
+
+import uk.gov.hmrc.domain.Nino
+import uk.gov.hmrc.individualincomedesstub.util.Validators
+
+import scala.util.matching.Regex
+import scala.util.matching.Regex.Match
+
+case class SelfAssessmentReturn(selfEmploymentStartDate: Option[String],
+                                saReceivedDate: String,
+                                selfEmploymentIncome: Double,
+                                employmentsIncome: Double)
+
+object SelfAssessmentReturn {
+  def apply(saReturnPayload: SelfAssessmentReturnRequestPayload): SelfAssessmentReturn = {
+    SelfAssessmentReturn(
+      saReturnPayload.selfEmploymentStartDate,
+      saReturnPayload.saReceivedDate,
+      saReturnPayload.selfEmploymentIncome.getOrElse(0.0),
+      saReturnPayload.employmentsIncome.getOrElse(0.0)
+    )
+  }
+}
+
+case class SelfEmployment(nino: Nino,
+                          taxYear: TaxYear,
+                          saReturns: Seq[SelfAssessmentReturn])
+
+case class SelfAssessmentReturnRequestPayload(selfEmploymentStartDate: Option[String],
```

Comment:
I think SelfAssessmentCreateRequest and SelfAssessmentReturnCreateRequest may be better?

Reply:
As it is not a request maybe SelfAssessmentReturnData